City-of-Helsinki/palvelutarjotin-ui
# PT-1723 | fix: getAllMenuPages, footer CMS page links, remove obsolete menus

## Description :sparkles:

**NOTE**:
 - This will **only** fetch **CMS pages** that are in a **menu** in some language or that are a translation of a page that is in a menu

### fix: getAllMenuPages, footer CMS page links, remove obsolete menus

fixes:
 - getAllMenuPages was using an obsolete menu, changed it to use the
   new menus after HDS 3 upgrade in PR #304 i.e.
   commit c64b314d5b4bc2f4a95251c5cb158032b2f85dad in master branch
 - footer CMS page links were broken, fixed them to use the /cms-page/
 - getAllMenuPages wasn't fetching footer CMS pages at all, added them

also:
 - use AppConfig.cmsGraphqlEndpoint where applicable
 - fix name passed to getEnvOrError in AppConfig.cmsGraphqlEndpoint

refs PT-1723

## Issues :bug:

### Closes :no_good_woman:

### Related :handshake:

[PT-1723](https://helsinkisolutionoffice.atlassian.net/browse/PT-1723)

## Testing :alembic:

Footer is empty in review environment, probably a configuration issue.
You can try locally running the application to see the footer in action:
 - checkout the PR's branch
 - set `NEXT_PUBLIC_CMS_BASE_URL=https://kultus.content.api.hel.fi/graphql` in `.env.local` to use production CMS
 - yarn
 - `yarn dev` or try building it `yarn build` as that'll use the `getAllMenuPages`

### Automated tests :gear:️

### Manual testing :construction_worker_man:

## Screenshots :camera_flash:

### Saavutettavuusseloste -link in Footer
![image](https://github.com/City-of-Helsinki/palvelutarjotin-ui/assets/77663720/e2f9b7fe-361d-410f-859f-4abfd794ec8c)

### Käyttöehdot -link in Footer
![image](https://github.com/City-of-Helsinki/palvelutarjotin-ui/assets/77663720/4a8ab633-ece4-4182-93a3-29a3c2c838c6)

## Additional notes :spiral_notepad:

[PT-1723]: https://helsinkisolutionoffice.atlassian.net/browse/PT-1723?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
